nos/common/spec.py

```diff
@@ -151,22 +149,42 @@ class FunctionSignature:
     method: str
     """Class method name. (e.g. forward, __call__ etc)"""
 
-    init_args: Tuple[Any, ...] = field(default_factory=tuple)
```

Comment:
[nit] might be good to validate `method != None`? same for `func_or_cls`

Reply:
That is already handled in the new pydantic validation